### DISCUSSION TITLE
Issue with Terraform State Files and Deployment Using Azure DevOps

### DISCUSSION THREAD

**Discussion by reshma67** [microsoft/PubSec-Info-Assistant#868] [General]
I've made several changes to the app, including adding OpenAI Mini and some UI-level updates, which were tested successfully in the local environment. However, during deployment, I accidentally lost the Terraform state files (they were added to .gitignore and my old machine crashed). Is there an alternative way to deploy the app to the same web app where the previous changes were made, using Azure DevOps?

**Comment by bjakems** [comment]
In the future, I recommend deploying from GitHub Codespaces so your data is not lost.  Please see the document [here](https://github.com/microsoft/PubSec-Info-Assistant/blob/main/docs/deployment/deployment.md#development-environment-configuration).  However, it is possible to import your state files from existing infrastructure with Terraform.  This will not recover your lost changes to your Terraform or webapp files if modified and not pushed to your repository.
Yes, it is possible to deploy with Azure DevOps, please see the documentation and files [here](https://github.com/microsoft/PubSec-Info-Assistant/blob/main/pipelines/README.md).
